### PR TITLE
Added emulation support 

### PR DESCRIPTION
**Pull request by fcasal**

### CODE REVIEW

File: src/Cutter.cpp [hunk 1]
```diff
@@ -730,65 +730,117 @@ void CutterCore::startDebug()
...
+        emit flagsChanged();
         currentlyDebugging = true;
     }
 }
 
+void CutterCore::startEmulation()
+{
+    if (!currentlyDebugging) {
+        offsetPriorDebugging = getOffset();
+    }
```
Comment (xarkes) [inline src/Cutter.cpp]: What if debugging is active?

File: src/Cutter.cpp [hunk 2]
```diff
@@ -730,65 +730,117 @@ void CutterCore::startDebug()
...
+    // clear registers, init esil state, stack, progcounter
+    cmd("ar0; aei; aeim; aeip");
+    emit registersChanged();
+    if (!currentlyDebugging || !currentlyEmulating) {
+        // prevent register flags from appearing during debug/emul
+        setConfig("asm.flags", false);
+        // consider adding io.cache = true to default config
+        // allows to view self-modifying code changes or other binary changes
+        setConfig("io.cache", true);
```
Comment (xarkes) [inline src/Cutter.cpp]: Maybe save the current value, in case someone tweak it by hand?
What do you think?

**Comment by xarkes** [comment]
Looks good!

File: src/Cutter.cpp (hunk 1 above)
Comment (fcasal) [reply src/Cutter.cpp]: We only save the offset before starting the debug/emulation.

Comment (xarkes) [reply src/Cutter.cpp]: I mean, can I start emulation while debugging has already started?

File: src/Cutter.cpp (hunk 2 above)
Comment (fcasal) [reply src/Cutter.cpp]: Sure, we can save them and restore after debug is finished

File: src/Cutter.cpp (hunk 1 above)
Comment (fcasal) [reply src/Cutter.cpp]: nope

Comment (xarkes) [reply src/Cutter.cpp]: Ok

**Pull request merged**